project_recalculate/models/project_task.py

```diff
@@ -193,39 +202,71 @@ def _interval_context_tz(self, interval):
         start = fields.Datetime.context_timestamp(self, start)
         end = datetime.now().replace(hour=interval[1])
         end = fields.Datetime.context_timestamp(self, end)
-        return (start.hour, end.hour)
+        result = Intervals()
+        result._items.append((start, end))
+        return result
 
     def _interval_default_get(self):
         default = (8, 18)
         return self._interval_context_tz(default)
 
+    def _resource_timezone(self, dt, resource=None, calendar=None):
+        if resource or calendar:
+            tz = timezone((resource or calendar).tz)
+        else:
+            tz = utc
+        return tz.localize(dt)
+
+    def _get_work_intervals(self, day_date, resource=None, calendar=None):
+        start_dt = datetime.combine(day_date, datetime.min.time())
+        end_dt = datetime.combine(day_date, datetime.max.time())
+
+        if not start_dt.tzinfo:
+            start_dt = self._resource_timezone(start_dt, resource)
+        if not end_dt.tzinfo:
+            end_dt = self._resource_timezone(end_dt, resource)
+
+        if not calendar:
+            intervals = self._interval_default_get()
+            intervals -= self._leave_intervals(start_dt, end_dt, resource)
+        else:
+            intervals = calendar._work_intervals(start_dt, end_dt, resource)
+
+        return intervals
+
     def _first_interval_of_day_get(self, day_date, resource=None,
                                    calendar=None):
-        default_interval = self._interval_default_get()
-        intervals = calendar.get_working_intervals_of_day(
-            start_dt=day_date, compute_leaves=True, resource_id=resource.id,
-            default_interval=default_interval,
-        )
-        return intervals and intervals[0] or False
-
-    def _calendar_schedule_days(self, days, day_date,
-                                resource=None, calendar=None):
+        intervals = self._get_work_intervals(day_date, resource, calendar)
+        return intervals._items and intervals._items[0] or False
```

Comment:
Isn't the first element accessible without needing to access protected variable? (`intervals[0]`?)